pyuvdata/uvdata.py

```diff
@@ -1774,3 +1775,147 @@ def antpairpol_iter(self, squeeze='default'):
         antpairpols = self.get_antpairpols()
         for key in antpairpols:
             yield (key, self.get_data(key, squeeze=squeeze))
+
+
+    def parse_ants(self,ant_str):
+        """
+        Generates two lists of antenna pair tuples and polarization indices based on
+        parsing of the string ant_str.  If no valid polarizations or antenna numbers
+        are found in ant_str (Stokes params, or combinations of [lr] or [xy]),
+        ant_pairs_nums and polarizations are returned as None.
+
+        Args:
+            ant_str: String containing antenna information to pass to select function.
+                Can be all, auto, cross, or combinations of antenna numers and polarization
+                indicators l and r or x and y.
+
+        Output:
+            ant_pairs_nums: List of tuples containing the parsed pairs of antennae numbers.
+                If 'all' is passed as ant_str, returned as None.
+            polarizations: List of desired polarizations.  If no polarizations found in ant_str
+                then returned as None.
+        """
+
+        ant_re = r'(\(((-?\d+[lrxy]?,?)+)\)|-?\d+[lrxy]?)'
+        bl_re = '(^(%s_%s|%s),?)' % (ant_re, ant_re, ant_re)
+        str_pos = 0
+        ant_pairs_nums = []
+        polarizations = []
+        ant_nums = self.get_ants()
+
+        while str_pos < len(ant_str):
+            m = re.search(bl_re, ant_str[str_pos:])
+            if m is None:
+                if ant_str[str_pos:].startswith('all'):
+                    pass
+                elif ant_str[str_pos:].startswith('auto'):
+                    for ant in ant_nums:
+                        ant_tuple = tuple((ant,ant))
+                        ant_pairs_nums.append(ant_tuple)
+                elif ant_str[str_pos:].startswith('cross'):
+                    for ant1 in ant_nums:
```

Comment:
This goes back to the discussion from Slack regarding how much the parser should know about the object.  I previously had implemented a check that an antenna pair would not be output in `ant_pairs_nums` unless it was present in `self.get_antpairs()`.  Should I put this check back in?  Or should that check be left to the select function?